src/main/java/mate/jdbc/dao/impl/ManufacturerDaoImpl.java

```diff
@@ -0,0 +1,115 @@
+package mate.jdbc.dao.impl;
+
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.Statement;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+import java.util.zip.DataFormatException;
+import mate.jdbc.dao.ManufacturerDao;
+import mate.jdbc.lib.Dao;
+import mate.jdbc.model.Manufacturer;
+import mate.jdbc.util.ConnectionUtil;
+
+@Dao
+public class ManufacturerDaoImpl implements ManufacturerDao {
+    @Override
+    public Manufacturer create(Manufacturer manufacturer) throws DataFormatException {
+        String insertManufacturerRequest = "INSERT INTO manufacturers(name,country) values(?,?);";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement createManufacturerStatement = connection
+                        .prepareStatement(insertManufacturerRequest,
+                        Statement.RETURN_GENERATED_KEYS)) {
+            createManufacturerStatement.setString(1,manufacturer.getName());
+            createManufacturerStatement.setString(2,manufacturer.getCountry());
+            createManufacturerStatement.executeUpdate();
+            ResultSet generatedKeys = createManufacturerStatement.getGeneratedKeys();
+            if (generatedKeys.next()) {
+                Long id = generatedKeys.getObject(1,Long.class);
+                manufacturer.setId(id);
+            }
+        } catch (SQLException e) {
+            throw new DataFormatException("Can't insert car to DB");
+        }
+        return manufacturer;
+    }
+
+    @Override
+    public Optional<Manufacturer> get(Long id) throws DataFormatException {
+        String getManufacturerRequest = "SELECT * FROM manufacturers"
+                + " where is_deleted = false and id = ?;";
+        try (Connection connection = ConnectionUtil.getConnection();
+                 PreparedStatement getManufacturerStatement = connection
+                        .prepareStatement(getManufacturerRequest)) {
+            getManufacturerStatement.setLong(1,id);
+            ResultSet resultSet = getManufacturerStatement.executeQuery();
+            resultSet.next();
+            String name = resultSet.getString(2);
+            String country = resultSet.getString(3);
+            Manufacturer manufacturer = new Manufacturer();
+            manufacturer.setName(name);
+            manufacturer.setCountry(country);
+            manufacturer.setId(id);
+            return Optional.of(manufacturer);
+        } catch (SQLException e) {
+            throw new DataFormatException("DB hasn't manufacturer with id " + id + ".");
```

Comment:
```suggestion
            throw new DataFormatException("DB can't get manufacturer with id " + id + ".");
```
DB may contain this model, but we got Sql exception since we wrote incorrect request